src/net/sourceforge/kolmafia/DebugModifiers.java

```diff
@@ -55,32 +55,24 @@ private static void flushRow() {
   }
 
   @Override
-  public void add(final int index, final double mod, final String desc) {
+  public void add(final int index, final double mod, final Lookup lookup) {
     if (index < 0 || index >= Modifiers.DOUBLE_MODIFIERS || mod == 0.0) {
       return;
     }
 
-    super.add(index, mod, desc);
+    super.add(index, mod, lookup);
 
     Integer key = index;
     if (!DebugModifiers.wanted.containsKey(key)) {
       return;
     }
 
-    String type;
-    String name;
-    int ind = desc.indexOf(":");
-    if (ind > 0) {
-      type = desc.substring(0, ind);
-      name = desc.replace(type + ":", "");
-    } else {
-      type = "";
-      name = desc;
-    }
-    if (!desc.equals(DebugModifiers.currentDesc) || DebugModifiers.adjustments.containsKey(key)) {
+    ModifierType type = lookup.type;
+    String name = lookup.getName();
+    if (!lookup.equals(DebugModifiers.currentDesc) || DebugModifiers.adjustments.containsKey(key)) {
```

Comment:
currentDesc should probably be renamed currentLookup, or possibly currentName.

Should this be a comparison against `name`, since you set currentDesc to `name` below?

Reply:
Fixed.